### ISSUE TITLE
vipsthumbnail - converting a cmyk image and exporting using ICC gives wrong colors

### ISSUE THREAD

**Issue by zvikam** [jcupitt/libvips#510]
When I try to convert a CMYK image and use a new ICC profile for exporting

    vipsthumbnail --vips-info -s 4288x2848 -e sRGB_IEC61966-2-1_black_scaled.icc source.jpg

The resulting image is brighter than the original (similar colors, just noticeably brighter).

The specified size is equal to original, so no resize.
The ICC profile is the standard one found online at [http://www.color.org/srgbprofiles.xalter](url)
The operation is successful, no errors or warnings:

    info: vipsthumbnail: thumbnailing source.jpg
    info: vipsthumbnail: selected loader is VipsForeignLoadJpegFile
    info: vipsthumbnail: loading jpeg with factor 1 pre-shrink
    info: vipsthumbnail: importing with embedded profile
    info: vipsthumbnail: converting to processing space srgb
    info: resize: box shrink by 1 x 1
    info: resize: residual affine 1 x 1
    info: resize: nearest interpolation
    info: vipsthumbnail: exporting to device space with a profile
    info: vipsthumbnail: thumbnailing source.jpg as tn_source.jpg

Source image: [https://drive.google.com/file/d/0B53nio2u3kjNZGZUWDlJeV9hQ3M/view?usp=sharing](url)

Am I doing something wrong or is this a problem?

**Comment by jcupitt** [comment]
Hi @zvikam, as far as I can see, it's working as intended. `vipsthumbnail` is transforming the image using LCMS with the embedded profile and your sRGB profile, using relative rendering intent. 

When you say `vipsthumbnail` is too bright, what are you comparing it to?

**Comment by zvikam** [comment]
Hi @jcupitt,
I'm comparing it to the original image.
Other ICC transforms I've done do not exhibit this difference in brightness, only my CMYK images do.

**Comment by jcupitt** [comment]
How are you viewing the original image? Your viewer program is converting the CMYK jpg to RGB for your screen, do you know what settings it is using?

I tried converting your image with relative and perceptual rendering intent, and got this:

http://www.rollthepotato.net/~john/icc/

```
$ vips icc_transform cmyk.jpg percep.jpg sRGB_IEC61966-2-1_black_scaled.icc --embedded --intent perceptual
$ vips icc_transform cmyk.jpg rel.jpg sRGB_IEC61966-2-1_black_scaled.icc --embedded --intent relative
```

Does the difference between `percep.jpg` and `rel.jpg` look like the difference you are seeing?

**Comment by zvikam** [comment]
I'm viewing the images both in chrome and showfoto.
`percep.jpg` looks a little better than `rel.jpg`, but both are visibly different than the original image.

**Comment by jcupitt** [comment]
I don't think chrome displays CMYK jpg images correctly, they come out very green, at least for me. I don't know `showfoto`, but if it looks like Chrome, it's probably also wrong. I would guess they are using a very simple formula for conversion and not looking at the profile. 

I see this:

http://www.rollthepotato.net/~john/shot.png

Chrome on the right, vips percep on the left. You can see the version on the right is not correct. 

Did you try photoshop or gimp?

**Comment by zvikam** [comment]
I was so concentrated on the red ceiling, I never noticed the walls are green :-)

Anyway, I see the same "green" walls in Chrome, showfoto and gimp, but not in google-drive preview...
And in google-drive preview the vips percep image look identical!
I wonder what they're doing right that everyone else isn't...

Thank you very much for your help!

**Comment by jcupitt** [comment]
gimp should work, perhaps only recent versions have fixed the colour management? 2.8.16 is identical to vips percep, for me.

**Comment by jcupitt** [comment]
Anyway, if this is resolved, please close the issue.
